### PR TITLE
[REVIEW] Export NVTabular + HugeCTR Ensemble Config

### PR DESCRIPTION
**Pull request by oyilmaz-nvidia**
This PR includes the functions that generates NVTabular + HugeCTR ensemble config.

### CODE REVIEW

File: nvtabular/inference/triton/__init__.py [hunk 1]
```diff
@@ -70,86 +70,96 @@ def export_tensorflow_ensemble(model, workflow, name, model_path, label_columns,
...
+    hugectr_params:
+        HugeCTR specific parameters
+    name:
+        The base name of the various triton models
+    output_path:
+        The path where the models will be served
+    label_columns:
+        Labels in the dataset (will be removed from the workflow)
+    version:
+        The version of the mode
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:109]:
```suggestion
        The version of the model
```

File: nvtabular/inference/triton/__init__.py [hunk 2]
```diff
@@ -250,6 +285,183 @@ def _generate_model_config(
...
+        hugectr_step.output_map[output_col.name] = output_col.name
+
+    config.ensemble_scheduling.step.append(nvt_step)
+    config.ensemble_scheduling.step.append(hugectr_step)
+
+    with open(os.path.join(output_path, "config.pbtxt"), "w") as o:
+        text_format.PrintMessage(config, o)
+    return config
+
+
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:339]: This function doesn't seem to be used anywhere - can we delete ?

```suggestion
```

File: nvtabular/inference/triton/__init__.py [hunk 3]
```diff
@@ -250,6 +285,183 @@ def _generate_model_config(
...
+        )
+
+    config.instance_group.append(model_config.ModelInstanceGroup(gpus=[0], count=1, kind=1))
+
+    config_hugectr = model_config.ModelParameter(string_value=hugectr_params["config"])
+    config.parameters["config"].CopyFrom(config_hugectr)
+
+    gpucache_val = "true"
+    if "gpucache" in hugectr_params.keys():
+        gpucache_val = hugectr_params["gpucache"]
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:395]: Use the dict 'get' method here:
```suggestion
    gpucache_val = hugectr_params.get("gpucache", "true")
```

File: nvtabular/inference/triton/__init__.py [hunk 4]
```diff
@@ -250,6 +285,183 @@ def _generate_model_config(
...
+
+    gpucache = model_config.ModelParameter(string_value=gpucache_val)
+    config.parameters["gpucache"].CopyFrom(gpucache)
+
+    gpucacheper_val = "0.5"
+    if "gpucacheper" in hugectr_params.keys():
+        gpucacheper_val = str(hugectr_params["gpucacheper_val"])
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:402]:
```suggestion
    gpucacheper_val = str(hugectr_params.get("gpucacheper_val", "0.5"))
```

File: nvtabular/inference/triton/__init__.py [hunk 5]
```diff
@@ -250,6 +285,183 @@ def _generate_model_config(
...
+    config.parameters["max_nnz"].CopyFrom(max_nnz)
+
+    embedding_vector_size = model_config.ModelParameter(
+        string_value=str(hugectr_params["embedding_vector_size"])
+    )
+    config.parameters["embedding_vector_size"].CopyFrom(embedding_vector_size)
+
+    embeddingkey_long_type_val = "true"
+    if "embeddingkey_long_type" in hugectr_params.keys():
+        embeddingkey_long_type_val = hugectr_params["embeddingkey_long_type"]
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:433]:
```suggestion
    embeddingkey_long_type_val = hugectr_params.get("embeddingkey_long_type",  "true")
```

File: nvtabular/inference/triton/__init__.py [hunk 6]
```diff
@@ -70,86 +70,96 @@ def export_tensorflow_ensemble(model, workflow, name, model_path, label_columns,
     tf_config = _generate_tensorflow_config(model, name + "_tf", tf_path)
 
     # generate the triton ensemble
-    ensemble_path = os.path.join(model_path, name)
+    ensemble_path = os.path.join(model_path, name + "_ens")
```
Comment (benfred) [inline nvtabular/inference/triton/__init__.py:73]: I'd rather not change the TF ensemble name if possible (even if it turns out we don't have a choice for hugectr).

**Review by benfred — CHANGES_REQUESTED** [review]
Thanks for getting this in!  

Can you run black on your code? We have a precommit hook that might help out here by running this automatically. You can enable by running this in the root folder of your local nvtabular repo:

```
pip install pre-commit
pre-commit install
```

File: nvtabular/inference/triton/__init__.py (hunk 2 above)
Comment (oyilmaz-nvidia) [reply nvtabular/inference/triton/__init__.py:339]: Yes, we don't use it. Deleting it. I wonder why flake8 didn't catch this :)

File: nvtabular/inference/triton/__init__.py (hunk 6 above)
Comment (oyilmaz-nvidia) [reply nvtabular/inference/triton/__init__.py]: Cool, just updated it.

**Review by rnyak — APPROVED** [review]

**Review by benfred — APPROVED** [review]

**Pull request merged**